apps/site/assets/ts/helpers/departureInfo.tsx

```diff
@@ -123,11 +135,54 @@ const isCommuterRail = (departureInfo: DepartureInfo): boolean => {
   return departureInfo?.routeMode === "commuter_rail";
 };
 
+const DefaultWrapper: React.FunctionComponent<unknown> = ({ children }) => (
+  <li>{children}</li>
+);
+/**
+ * From a list of `DepartureInfo[]`, generate a list of `<DisplayTime />`.
+ * Each `<DisplayTime />` will recieve the `isCR` and `targetDate` props.
+ *
+ * This function can optionally
+ * - wrap each `<DisplayTime />` in a specified wrapper element, `WrapperEl`
+ * - truncate the list to a desired `listLength` value
+ */
+const departuresListFromInfos = (
+  departureInfos: DepartureInfo[],
+  isCR: boolean,
+  targetDate?: Date,
+  listLength?: number,
+  WrapperEl: typeof DefaultWrapper = DefaultWrapper
+): React.ReactElement[] => {
+  // optional cutoff time, before which we won't show schedules.
+  // just used with subway for now.
+  const predictionTimeCutoff = chain(departureInfos)
+    .filter(d => d.routeMode === SUBWAY)
+    .maxBy("prediction.time")
+    .value()?.prediction?.time;
```

Comment:
nit-pick(non-blocking): Since we are maxing by `"prediction.time"` I would think the `prediction` and `time` field would be defined, so they could probably be `prediction!.time`;  Though not sure if an `!` is allowed after a `?`

Reply:
I think you're right, and I just tried it and it works.